src/openrct2/world/Duck.cpp

```diff
@@ -88,7 +88,7 @@ rct_duck * rct_sprite::AsDuck()
     rct_duck * result = nullptr;
     if (IsDuck())
     {
-        return (rct_duck *)result;
+        return result;
```

Comment:
How does this even work?

Reply:
should be `this`

Reply:
Oh my

Reply:
Why did no one notice ducks weren't working properly?

Reply:
`return (🦆 *) this`

Reply:
This will need network version change

Reply:
@IntelOrca It doesn't look like this function is called anywhere yet.

Reply:
Well, I'll be ducked.